src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -149,6 +154,96 @@ public final EventLoop eventLoop() {
     }
 
     /**
+     * Returns a {@link EventExecutor} that will make sure this invocation is set
+     * as the current invocation before executing any callback. This should
+     * almost always be used for executing asynchronous callbacks in service
+     * code to make sure features that require the invocation context work
+     * properly. Most asynchronous libraries like
+     * {@link java.util.concurrent.CompletableFuture} provide methods that
+     * accept an {@link Executor} to run callbacks on.
+     */
+    public final EventExecutor contextAwareEventExecutor() {
+        return new ServiceInvocationContextAwareEventExecutor(this, eventLoop());
+    }
+
+    public final Executor makeContextAware(Executor executor) {
+        return runnable -> executor.execute(makeContextAware(runnable));
+    }
+
+    public final <T> Callable<T> makeContextAware(Callable<T> callable) {
+        ServiceInvocationContext propagatedContext = this;
+        return () -> {
+            boolean mustResetContext = propagateContextIfNotPresent(propagatedContext);
+            try {
+                return callable.call();
+            } finally {
+                if (mustResetContext) {
+                    removeCurrent();
+                }
+            }
+        };
+    }
+
+    public final Runnable makeContextAware(Runnable runnable) {
+        ServiceInvocationContext propagatedContext = this;
+        return () -> {
+            boolean mustResetContext = propagateContextIfNotPresent(propagatedContext);
+            try {
+                runnable.run();
+            } finally {
+                if (mustResetContext) {
+                    removeCurrent();
+                }
+            }
+        };
+    }
+
+    public final <T extends Future<?>> GenericFutureListener<T> makeContextAware(
```

Comment:
From what I can tell everything has parameters, can you point out what you mean?